precompiles/xcm/src/lib.rs

```diff
@@ -266,7 +277,17 @@ where
             .iter()
             .cloned()
             .filter_map(|address| {
-                R::address_to_asset_id(address.into()).and_then(|x| C::reverse_ref(x).ok())
+                let address: H160 = address.into();
+
+                // Special case for native asset
+                if address == H160::from([0xff; 20]) {
```

Comment:
`H160::MAX` is already used for relay chain native asset.

You should use `MAX -1`. Or some other perhaps, so they don't just differ by 1 byte.